pylint/checkers/base.py

```diff
@@ -1467,6 +1472,19 @@ def visit_dict(self, node: nodes.Dict) -> None:
                 self.add_message("duplicate-key", node=node, args=key)
             keys.add(key)
 
+    @utils.check_messages("duplicate-value")
+    def visit_set(self, node: nodes.Set) -> None:
+        """Check duplicate value in set."""
+        values = set()
+        for v in node.elts:
+            if isinstance(v, nodes.Const):
+                value = v.value
+            else:
```

Comment:
Could we catch some more cases?

Matching `nodes.Name` would pick up:

```
a = 1
b = 2
s = {a, b, b}
```

This would be really useful. Another case could be Tuples of constants, e.g. `{(12, 15), (12, 15), (12)}`.

Reply:
Instead of opting in for certain types, what about opting out instead? What kinds of things should be ignored?

Reply:
@ksaketou This comes down to you wanting to put more time in. I would be fine with merging this as a very bare-bones checker and expanding later.